composer/utils/collect_env.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Helpers to gather system information for debugging and bug reporting.
+
+Leverages PyTorch's :mod:`torch.utils.collect_env` package to gather pertinent system information.
+The following information is additionally collected to faciliate Comopser specific debug:
+
+* Composer version
+* Number of nodes
+* Host processor model name
+* Host processor physical core count
+* Number of accelerators per node
+* Accelerator model name
+
+This package can be invoked as a standalone console script or can be invoked from within
+an application to gather and generate a system environment report.
+
+The module can be invoked by using the entrypoint alias:
+
+.. code-block::
+
+    composer_collect_env
+
+Or manually as a standalone script:
+
+.. code-block::
+
+    python composer/utils/collect_env.py
+
+To generate a system report from within a user application see :func:`print_env`.
+
+A custom excepthook wrapper is also provided which extends the default :func:`sys.excepthook`
+to automatically collect system information when an exception is raised.
+
+To override the default :func:`sys.excepthook` see :func:`configure_excepthook`.
+
+To restore the default :func:`sys.excepthook` see :func:`restore_excepthook`.
+"""
+
+import sys
+import time
+from typing import NamedTuple
+
+import cpuinfo
+import psutil
+
+__all__ = ["configure_excepthook", "print_env", "restore_excepthook"]
+
+# Check if PyTorch is installed
+try:
+    import torch.utils.collect_env as torchenv
+    from torch.cuda import get_device_name as accel_device_name
+    from torch.cuda import is_available as cuda_available
+    TORCH_AVAILABLE = True
+except (ImportError, NameError, AttributeError, OSError):
```

Comment:
Would it work to just catch the `ImportError`? Or do we need to catch these other errors?

Reply:
`ImportError` should work, was just sort of blindly doing this to match the convention used in the Torch `collect_env` script but it might be overly conservative:
https://github.com/pytorch/pytorch/blob/2f2158ae4502667951a4d6f5f95d751c7d0927d8/torch/utils/collect_env.py#L18